spec/requests/api/v3/work_packages/work_packages_by_project_resource_spec.rb

```diff
@@ -0,0 +1,132 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+
+require 'spec_helper'
+require 'rack/test'
+
+describe API::V3::WorkPackages::WorkPackagesByProjectAPI, type: :request do
+  include Rack::Test::Methods
+  include API::V3::Utilities::PathHelper
+
+  let(:project) { FactoryGirl.create(:project_with_types, id: 5, is_public: false) }
+
+  describe '#post' do
+    let(:status) { FactoryGirl.build(:status, is_default: true) }
+    let(:priority) { FactoryGirl.build(:priority, is_default: true) }
+    let(:parameters) { { subject: 'new work packages' } }
+    let(:permissions) { [:add_work_packages, :view_project] }
+    let(:role) { FactoryGirl.create(:role, permissions: permissions) }
+    let(:current_user) do
+      FactoryGirl.build(:user, member_in_project: project, member_through_role: role)
+    end
+    let(:path) { api_v3_paths.work_packages_by_project project.id }
+
+    before do
+      status.save!
+      priority.save!
+
+      allow(User).to receive(:current).and_return current_user
+      ActionMailer::Base.deliveries.clear
```

Comment:
If it does not work any other way keep this line in. But I still don't believe it is neccessary, because the `user_mailer_spec.rb` does never explicitly clear the deliveries and the tests are quite stable.